src/lib/parallel_scan/parallel_scan.ml

```diff
@@ -840,6 +910,30 @@ module T = struct
       end
 
       include T
+
+      let with_leaner_trees ({trees; _} as t) =
```

Comment:
What are you throwing out here? What's the overhead w.r.t. just storing the data in a flat buffer?

Reply:
All the merge jobs that are done (with `Job_status.Done`) since we don't need those anywhere as of now. 
You don't have to construct the trees again? There's going to be `(transaction_capacity_log_2 + 1) * (delay + 1)` number of trees

Reply:
I guess I am not sure what is the additional overhead in bin-io-ing a tree vs a flat buffer? The number of "actual" nodes are still the same. The discarded jobs will be replaced with the `Empty` constructor

Reply:
I'm thinking in terms of raw bytes, if we have a 2^9 scan state for example, how many bytes will this take up. And how many bytes will just having the data in a flat buffer take up -- what is the overhead per entry. (This has bandwidth implications)

Reply:
As discussed this should be okay? Just for the record, because of the type of the tree here, there will only be 9 (or how much ever is the depth of the tree) extra bytes per tree for the constructors of the tree.